expenses-manager-frontend/src/main/frontend/src/app/daily-stats/daily-stats.component.html

```diff
@@ -1,22 +1,31 @@
 <div class="dashboard-container">
-    <div class="chart-container">
-      <ngx-charts-pie-chart
-        [view]="[300, 300]"
-        [results]="data"
-        [legend]="false"
-        [doughnut]="true"
-        [labels]="false"
-        [scheme]="colorScheme">
-      </ngx-charts-pie-chart>
-      <div class="legend-container">
-        <ul>
-          <li><span class="legend-color" style="background-color: #46B1D9;"></span> Food & Restaurants</li>
-          <li><span class="legend-color" style="background-color: #5844E3;"></span> Car</li>
-          <li><span class="legend-color" style="background-color: #E047AB;"></span> Subscriptions</li>
-          <li><span class="legend-color" style="background-color: #F78486;"></span> Entertainment</li>
-          <li><span class="legend-color" style="background-color: #A1D12E;"></span> Coffee</li>
-        </ul>
-      </div>
+
+  <!--
+    <mat-form-field appearance="fill">
+    <mat-label>Select Date</mat-label>
+    <input matInput [matDatepicker]="picker" (dateChange)="onDateChange($event)" placeholder="Choose a date">
+    <mat-datepicker #picker></mat-datepicker>
+  </mat-form-field>
+  -->
+
+
+  <div class="chart-container">
+    <ngx-charts-pie-chart
+      [view]="[300, 300]"
+      [results]="data"
+      [legend]="false"
+      [doughnut]="true"
+      [labels]="false"
+      [scheme]="colorScheme">
+    </ngx-charts-pie-chart>
+    <div class="legend-container">
+      <ul>
+        <li><span class="legend-color" style="background-color: #46B1D9;"></span> Food & Restaurants</li>
```

Comment:
Aici as crea o lista de obiecte in fisierul de typescript la componentei, iar fiecare obiect sa simuleze structura listei din baza de date pe care o vom procesa dupa ce vine pe request. Asadar, fiecare obiect va avea id, name, colorCode. Aici aici vom itera prin ele folosing ngFor si vom seta numele si culoarea corespunzatoare. In acest fel nu va mai trebui sa hardcodam coduri de culoare specifice pentru o anumita categorie cum ai facut in fisierul styles.css sau in acest fisier de scss cu selector de copii (.legend-color:nth-child(1)) Pentru ca nu vesi stii niciodata cate date iti vor veni si care sunt acestea. Trebuie sa fim cat mai generici posibili .